core/server/master/src/main/java/alluxio/master/metrics/CloudCostUtils.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metrics;
+
+import alluxio.master.file.DefaultFileSystemMaster.Metrics.UFSOps;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * This class contains utilities to compute cloud cost savings.
+ */
+public final class CloudCostUtils {
+  private CloudCostUtils() {}  // prevent instantiation
+
+  public static final Map<UFSOps, Double> S3COSTMAP =
+      ImmutableMap.of(
+          UFSOps.CREATE_FILE, 0.000005,
+          UFSOps.GET_FILE_INFO, 0.000005,
+          UFSOps.DELETE_FILE, 0.0,
+          UFSOps.LIST_STATUS, 0.000005);
+  public static final Map<UFSOps, Double> ABFSCOSTMAP =
+      ImmutableMap.of(
+          UFSOps.CREATE_FILE, 0.0000065,
+          UFSOps.GET_FILE_INFO, 0.0000065,
+          UFSOps.DELETE_FILE, 0.0,
+          UFSOps.LIST_STATUS, 0.0000065);
+  public static final Map<UFSOps, Double> GCSCOSTMAP =
+      ImmutableMap.of(
+          UFSOps.CREATE_FILE, 0.000005,
+          UFSOps.GET_FILE_INFO, 0.0000004,
+          UFSOps.DELETE_FILE, 0.0,
+          UFSOps.LIST_STATUS, 0.000005);
+  public static final Map<UFSOps, Double> OSSCOSTMAP =
+      ImmutableMap.of(
+          UFSOps.CREATE_FILE, 0.000005,
+          UFSOps.GET_FILE_INFO, 0.0000004,
+          UFSOps.DELETE_FILE, 0.0,
+          UFSOps.LIST_STATUS, 0.000005);
+  public static final Map<String, Map<UFSOps, Double>> COSTMAP =
+      ImmutableMap.of(
+       "abfs", ABFSCOSTMAP,
+       "gcs", GCSCOSTMAP,
```

Comment:
Are the keys supposed to be the scheme? I think the gcs scheme is `gs`.

Also, I think sometimes s3 can have different schemes like `s3n`, or `s3a`. Is there a way we can just use the UFS factories to match the keys?

Reply:
keys are the getFsType associated with each connector. It is unique to each connector and covers `s3n` and `s3a` schemes.